swarm.sh

```diff
@@ -1,94 +1,152 @@
-
-    # Clickhouse
-    config::generate_service_configs cares-clickhouse-config-importer / "${COMPOSE_FILE_PATH}/importer/analytics-datastore-clickhouse/" "${COMPOSE_FILE_PATH}/importer" clickhouse
-    clickhouse_temp_compose_param="-c ${COMPOSE_FILE_PATH}/importer/docker-compose.tmp.yml"
-
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/importer/docker-compose.config.yml $clickhouse_temp_compose_param instant" "Failed to deploy Cares on Platform"
-
-    log info "Waiting to update configs"
-    # Kafka Mapper Consumer
-    REF_service_update_args=""
-    config::update_service_configs REF_service_update_args /app/src/data "$COMPOSE_FILE_PATH"/importer/kafka-mapper-consumer/mapping kafka-mapper-consumer
-    config::update_service_configs REF_service_update_args /app/src/plugin "$COMPOSE_FILE_PATH"/importer/kafka-mapper-consumer/plugin kafka-mapper-consumer
-    try "docker service update $REF_service_update_args instant_kafka-mapper-consumer" "Failed to update config for instant_kafka-mapper-consumer"
-    # Superset
-    REF_service_update_args=""
-    config::update_service_configs REF_service_update_args /app/pythonpath "$COMPOSE_FILE_PATH"/importer/dashboard-visualiser-superset superset
-    # TODO: Update .env.superset once the value for MAPBOX_API_KEY is known
-    config::env_var_add_from_file REF_service_update_args "$COMPOSE_FILE_PATH"/.env.superset
-    try "docker service update $REF_service_update_args instant_dashboard-visualiser-superset" "Failed to update config for instant_dashboard-visualiser-superset"
-
-    log info "Waiting to give config importers time to run before cleaning up service"
-    config::remove_config_importer cares-clickhouse-config-importer
-    config::remove_config_importer cares-superset-config-importer
-    config::remove_config_importer hapi-fhir-config-importer
-
-    # Ensure config importer is removed
-    config::await_service_removed instant_cares-clickhouse-config-importer
-    config::await_service_removed instant_cares-superset-config-importer
-    config::await_service_removed instant_hapi-fhir-config-importer
-    
-    log info "Restarting HAPI FHIR.."
-    try "docker service scale instant_hapi-fhir=0" "Error scaling down hapi-fhir to update the IG"
-    try "docker service scale instant_hapi-fhir=$HAPI_FHIR_INSTANCES" "Error scaling up hapi-fhir to update the IG"
-
-    log info "Removing stale configs..."
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "superset"
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "clickhouse"
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "hapi-fhir"
-
-    log info "Cleaning containers and configs"
+  log info "Removing stale configs..."
+  for config_label in "${CONFIG_LABELS[@]}"; do
+    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "$config_label"
```

Comment:
Do `prune_configs` and `remove_stale_service_configs` do different things (in effect). If they don't, then I think we should only use one. Preferably `prune_configs` since we can just pass through the array to that function

Reply:
Yes they are different, remove_stale_service_config will remove old configs and keep the new one, 
prune_configs willlremove all configs having the label given as a param

Reply:
But you can't remove a config from a service that is still around though even if it is scaled to zero instances, so are you sure that they don't do the same thing?

Reply:
That is also why it was given the name `prune_configs` as opposed to `remove_configs`

Reply:
I think the config importers will be removed completely and not just scaled to 0, so prune_configs will eventually remove these configs. 
To be honest not sure why keeping them ..

Reply:
Thinking about it more, I think it was like that to not update the configs if there is no change
because the digest is the same in case the config is the same